syntax = "proto3";

package audioalarm.v1;

enum CameraType {
	Unknown = 0;
	DaHua = 1;
	HiK = 2;
}

enum ResponseCode {
	Ok_200 = 0;                           // successed
	Err_Camera_Unsupported = 1;           // camera is not supported
	Err_Device_Not_Found = 2;             // device is not found
	Err_Device_Login_Incomplete = 3;      // user information is incomplete
	Err_Audio_File_Incomplete = 4;        // audio file information is incomplete
	Err_Play_Duration_Not_Set = 5;        // play duration is not set
	Err_500 = 6;                          // server inner error
}

// The request message
message AlarmRequest {
	uint32 msg_id = 1;
	CameraType camera_type = 2;
	string dev_ip = 3;
	uint32 dev_port = 4;
	string user_name = 5;
	string password = 6;
    string download_url = 7;
    string file_name = 8;
    string md5_value = 9;
	uint32 play_duration = 10;  // in units of seconds
}

// The response message
message AlarmReply {
	ResponseCode code = 1;
    string message = 2;
}

service AlarmService {
  // call alarm interface
  rpc SetAlarmMessage (AlarmRequest) returns (AlarmReply) {}
}
